### PR TITLE
 Move metadata from non-standard `setup.cfg` into `PEP 621`-compliant `pyproject.toml`.

### PR DESCRIPTION
**Pull request by KOLANICH**
Also fixes some metadata issues:
* license name is replaced with its SPDX id
* links to display in PyPI are fixed. The main link is now GitHub repo, and added a `Documentation` link

### CODE REVIEW

**Comment by mperrin** [comment]
@BradleySappington @obi-wan76 FYI we received this PR with suggested improvements to some of the packaging metadata for poppy. I'll let the two of you decide how to handle this.  We should also consider the same for `webbpsf` (and more broadly, presumably at some level we should do this consistently in all the INS python packaging...?)

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,57 @@
 [build-system]
-requires = ["setuptools >= 38.0.0",
-            "setuptools_scm >= 2.0.0",
+requires = ["setuptools >= 61.2",
+            "setuptools_scm[toml]>=3.4.3",
```
Comment (BradleySappington) [inline pyproject.toml]: Why did you choose these specific versions as minimums?

File: pyproject.toml [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
-build-backend = 'setuptools.build_meta'
\ No newline at end of file
+build-backend = 'setuptools.build_meta'
+
+[project]
+name = "poppy"
+description = "Physical optics propagation (wavefront diffraction) for optical simulations, particularly of telescopes."
+authors = [{name = "Marshall Perrin", email = "mperrin@stsci.edu"}]
+license = {text = "BSD-3-Clause"}
+requires-python = ">=3.8"
```
Comment (BradleySappington) [inline pyproject.toml]:
```suggestion
requires-python = ">=3.9"
```
new min version

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+dependencies = [
+    "numpy>=1.18.0",
```
Comment (BradleySappington) [inline pyproject.toml]:
```suggestion
    "numpy>=1.20.0",
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+    "scipy>=1.5.0",
+    "matplotlib>=3.2.0",
+    "astropy>=4.0.0",
```
Comment (BradleySappington) [inline pyproject.toml]:
```suggestion
    "astropy>=4.3.0",
```

**Review by BradleySappington — CHANGES_REQUESTED** [review]
Thanks for the PR, please make the below changes and update with latest dev.

File: pyproject.toml [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+Homepage = "https://github.com/spacetelescope/poppy"
+Documentation = "https://poppy-optics.readthedocs.io/"
+"Bug Tracker" = "https://github.com/spacetelescope/poppy/issues"
+
+[project.readme]
+text = '"POPPY (**P**\ hysical **O**\ ptics **P**\ ropagation in **Py**\ thon) is a Python package that simulates physical optical propagation including diffraction. It implements a flexible framework for modeling Fraunhofer and Fresnel diffraction and point spread function formation, particularly in the context of astronomical telescopes."'
+content-type = "text/markdown"
+
+[tool.setuptools]
+zip-safe = false
```
Comment (BradleySappington) [inline pyproject.toml]: I believe zip-safe is now obsolete.  Can this line be removed entirely?

File: pyproject.toml (hunk 1 above)
Comment (KOLANICH) [reply pyproject.toml]: `61.2` is a minimum version of `setuptools` when I felt support of `PEP 621` in it become usable. It got `PEP 621` a few versions earlier, but it used to have a few bugs.

`3.4.3` is the version used to be recommended in `setuptools_scm` docs. Now I have upgraded it to `7`. In `7` `setuptools_scm` got support of making sdists that can be built without git.

File: pyproject.toml (hunk 2 above)
Comment (KOLANICH) [reply pyproject.toml]: Fixed.

File: pyproject.toml (hunk 5 above)
Comment (KOLANICH) [reply pyproject.toml]: Fixed.

File: pyproject.toml (hunk 4 above)
Comment (KOLANICH) [reply pyproject.toml]: Fixed.

File: pyproject.toml [hunk 6]
```diff
@@ -1,5 +1,54 @@
...
-build-backend = 'setuptools.build_meta'
\ No newline at end of file
+requires = ["setuptools >= 61.2", "setuptools_scm[toml]>=7"]
+build-backend = 'setuptools.build_meta'
+
+[project]
+name = "poppy"
+description = "Physical optics propagation (wavefront diffraction) for optical simulations, particularly of telescopes."
+authors = [{name = "Marshall Perrin", email = "mperrin@stsci.edu"}]
+license = {text = "BSD-3-Clause"}
```
Comment (BradleySappington) [inline pyproject.toml:9]: use license file

```suggestion
license = {file = "LICENSE.md"}
```

File: setup.cfg [hunk 7]
```diff
@@ -1,49 +1,8 @@
-[metadata]
-name = poppy
-description = Physical optics propagation (wavefront diffraction) for optical simulations, particularly of telescopes.
-long_description = "POPPY (**P**\ hysical **O**\ ptics **P**\ ropagation in **Py**\ thon) is a Python package that simulates physical optical propagation including diffraction. It implements a flexible framework for modeling Fraunhofer and Fresnel diffraction and point spread function formation, particularly in the context of astronomical telescopes."
-author = Marshall Perrin
-author_email = mperrin@stsci.edu
-license = BSD
-license_file = LICENSE.md
-url = https://poppy-optics.readthedocs.io/
-edit_on_github = False
```
Comment (BradleySappington) [inline setup.cfg:10]: Will need to remove `Options for the edit_on_github extension` in docs/conf.py

File: setup.cfg [hunk 8]
```diff
@@ -1,49 +1,8 @@
...
-    sphinx-astropy
-    sphinx_rtd_theme
-    sphinx-automodapi
-    sphinx-issues
-
-[options.package_data]
-* = *.fits, *.csv
-poppy.tests = data/*
-
 [tool:pytest]
```
Comment (BradleySappington) [inline setup.cfg:6]: Add pytest to pyproject.toml using tag
`[tool.pytest.ini_options]`

**Review by BradleySappington — CHANGES_REQUESTED** [review]
Thanks for your updates.  A few more things I noticed on a more in depth review are outlined below.  

Will also need to update docs/conf.py to remove references to setup.cfg and to read from pyproject.toml instead of setup.conf

There are a few spots in there accessing setup.cfg,  but the main one is updating Project Information with something similar to the following:

`# -- Project information -----------------------------------------------------
 with open(Path(__file__).parent.parent / "pyproject.toml", "rb") as metadata_file:
     configuration = tomli.load(metadata_file)
     metadata = configuration['project']
     project = metadata['name']
     author = metadata["authors"][0]["name"]
     copyright = f'{datetime.now().year}, {author}'

     # The short X.Y version.
     try:
         version = project.__version__.split('-', 1)[0]
         # The full version, including alpha/beta/rc tags.
         release = project.__version__
     except AttributeError:
         version = 'dev'
         release = 'dev'`

**Comment by BradleySappington** [comment]
@KOLANICH - If you prefer I take over the PR and make the changes outlined above I'm happy to do so.  Please let me know your intentions by 17 APRIL 2023.  I appreciate your contributions thus far and would like to get it wrapped up.

**Comment by KOLANICH** [comment]
Oops, I'm sorry, I have forgotten to respond here. Of course, I'm completely OK with that.

**Comment by zacharyburnett** [comment]
@KOLANICH in case you are interested, [`peppyproject`](https://github.com/zacharyburnett/peppyproject) and [`ini2toml`](https://github.com/abravalheri/ini2toml/) seek to do this process automatically

**Comment by KOLANICH** [comment]
@zacharyburnett, thanks for the info. I personally used `ini2toml` while making this PR. (for converting `setup.py` into `setup.cfg` there is `setuptools_py2cfg`) Because it often needs some manual changes, it is likely that a bot doing this automatically and sending PRs is infeasible at current technology level.

**Pull request closed without merging**